### PR TITLE
Generate clients

### PR DESCRIPTION
**Pull request by vaikas**
This closes #

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes
Add codegen generated clients so that these can be used from things like github.com/knative-sandbox/eventing-rabbitmq. While it seems like a lot of new code, it's all code generated. Also had to fix couple of tags / add methods in the api/v1alpha2 so the code-gen behaves properly.

Start of the usage can be seen here (by no means complete, just starting that work over there):

https://github.com/knative-sandbox/eventing-rabbitmq/pull/267

### CODE REVIEW

File: hack/boilerplate.go.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/*
```
Comment (MirahImage) [inline hack/boilerplate.go.txt:1]: Does it make sense to use a separate file here rather than re-use NOTICE.go.txt

**Review by MirahImage — COMMENTED** [review]
Does the script `hack/update-codegen.sh` need to be added to the `make generate` target?

**Comment by n3wscott** [comment]
> Does the script `hack/update-codegen.sh` need to be added to the `make generate` target?

Does kubebuilder now encourage a Makefile? This has been seen as an anti-pattern in many projects, including Knative.

File: hack/boilerplate.go.txt (hunk 1 above)
Comment (vaikas) [reply hack/boilerplate.go.txt:1]: Yep, totes missed the NOTICE.go.txt, updated.

**Review by MirahImage — APPROVED** [review]

**Pull request merged**